cmd/buildah/push.go

```diff
@@ -98,16 +102,46 @@ func pushCmd(c *cobra.Command, args []string, iopts pushResults) error {
 		return errors.New("Only two arguments are necessary to push: source and destination")
 	}
 
-	compress := imagebuildah.Gzip
-	if iopts.disableCompression {
-		compress = imagebuildah.Uncompressed
-	}
-
 	store, err := getStore(c)
 	if err != nil {
 		return err
 	}
 
+	// Map of image tags found for the specified image.
+	// tags stored in the key
+	var imageTags map[string]string
+	if iopts.allTags {
+		imageTags = make(map[string]string)
+		if err := getAllTags(src, imageTags, store); err != nil {
+			return err
+		}
+		logrus.Debugf("For the image: [%s] found tags: %v", src, imageTags)
+	}
+
+	var errs *multierror.Error
+	var pushErr error
+	imageName := src
+	destTarget := destSpec
+	if len(imageTags) > 1 {
+		for tagKey := range imageTags {
+			src = imageName + ":" + tagKey
+			destSpec = destTarget + ":" + tagKey
+			pushErr = pushImage(c, src, destSpec, store, iopts)
```

Comment:
Nit: why not just use err?

Reply:
I could.  I think when I was first working on this, err was already in use, but I restructured.